embedded/sql/engine.go

```diff
@@ -1110,14 +1118,32 @@ func (e *Engine) ExecPreparedStmts(stmts []SQLStmt, params map[string]interface{
 
 	currTx := tx
 
+	changedDB := false
+
 	for _, stmt := range stmts {
 		if stmt == nil {
 			return nil, nil, ErrIllegalArguments
 		}
 
+		_, isDBSelectionStmt := stmt.(*UseDatabaseStmt)
+
+		if changedDB && e.multidbHandler != nil && isDBSelectionStmt {
```

Comment:
I don't understant this condition, why we're checking for `e.multidbHandler != nil` here?

Shouldn't `UseDatabaseStmt` always fail if there's no `e.multidbHandler` set ?

Reply:
the embedded still supports multiple databases, but if there is a multidbHandler set, the multi-database operations are delegated to it. 
But the selection of the database can not happen in the middle of a transaction

Reply:
If database can not be changed then why it's not this one?

```suggestion
		if changedDB {
```

(that flag would have to be then reset right after `currTx.commit()` few lines below).